api/test/logs/logger_test.cc

```diff
@@ -79,36 +87,65 @@ TEST(Logger, LogMethodOverloads)
                 opentelemetry::common::MakeAttributes({{"key1", "value 1"}, {"key2", 2}}));
   logger->Trace(m);
   logger->Trace(opentelemetry::common::MakeAttributes({{"key1", "value 1"}, {"key2", 2}}));
+  logger->Trace(trace_event_id, "Test log message",
+                opentelemetry::common::MakeAttributes({{"key1", "value 1"}, {"key2", 2}}));
```

Comment:
Just to add, if we are not using the variadic arguments in user-facing API, we can also add few more overloading to simplify the API by removing the use of `opentelemetry::common::MakeAttributes`. So something below should work:

```
logger->Trace(trace_event_id, "Test log message", {{"key1", "value 1"}, {"key2", 2}});
```

Reply:
There are at least 1 type of event id, 1 type of trace infomation, 1 type of timestamp, 1 type of body, 3 types of attributes, 6 different severities and a `Log` function with severity parameter by noww.
Every field is optional, then we have to write `2*2*2*2*4*7=448` functions for every paramenters.
Do you think this will also confuse users and also make us hard to maintain if we add more special attributes in the future?

Reply:
Yeah good point. This will indeed increase number of overloads.